app/main.py

```
@@ -1,5 +1,72 @@
+
+
 class Cargo:
     def __init__(self, weight: int) -> None:
         self.weight = weight
 
-# write your code here
+
+class BaseRobot:
+    def __init__(self,
+                 name: str,
+                 weight: int,
+                 coords: Optional[List[int]] = None
+                 ) -> None:
+        self.name = name
+        self.weight = weight
+        self.coords = coords if coords is not None else [0, 0]
+
+    def go_forward(self, step: int = 1) -> None:
+        self.coords[1] += abs(step)
+
+    def go_back(self, step: int = 1) -> None:
+        self.coords[1] -= abs(step)
+
+    def go_right(self, step: int = 1) -> None:
+        self.coords[0] += abs(step)
+
+    def go_left(self, step: int = 1) -> None:
+        self.coords[0] -= abs(step)
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(self,
+                 name: str,
+                 weight: int,
+                 coords: Optional[List[int]] = None
+                 ) -> None:
+        super().__init__(name,
+                         weight,
+                         coords if coords is not None else [0, 0, 0]
+                         )
+
+    def go_up(self, step: int = 1) -> None:
+        self.coords[2] += abs(step)
+
+    def go_down(self, step: int = 1) -> None:
+        self.coords[2] -= abs(step)
+
+
+class DeliveryDrone(FlyingRobot):
+    def __init__(self,
+                 name: str,
+                 weight: int,
+                 max_load_weight: int,
+                 coords: Optional[List[int]] = None,
+                 current_load: Cargo | None = None
+                 ) -> None:
+        super().__init__(name, weight, coords)
+        self.max_load_weight = max_load_weight
+        self.current_load = current_load
+
+    def hook_load(self, cargo: Cargo) -> None:
+        if self.current_load is None and cargo.weight <= self.max_load_weight:
```

The condition checks if the current load is None and if the cargo weight is within the max load weight. However, if the cargo weight exceeds the max load weight, it might be better to raise an exception or log a message instead of just passing.